### PR TITLE
fix the type definition of namer function

### PR DESCRIPTION
**Pull request by vim345**
Related issue: https://github.com/python/typeshed/issues/1497#issuecomment-317883223

### CODE REVIEW

File: stdlib/2and3/logging/handlers.pyi [hunk 1]
```diff
@@ -29,7 +29,7 @@ class WatchedFileHandler(Handler):
 
 if sys.version_info >= (3,):
     class BaseRotatingHandler(FileHandler):
-        namer = ...  # type: Optional[Callable[[str], None]]
+        namer = ...  # type: Callable[[str], str]
```
Comment (JelleZijlstra) [inline stdlib/2and3/logging/handlers.pyi]: Why did you remove the `Optional`? From https://github.com/python/cpython/blob/master/Lib/logging/handlers.py it looks like it is indeed allowed to be None.

Comment (vim345) [inline stdlib/2and3/logging/handlers.pyi]: I added back the Optional.

**Comment by vim345** [comment]
@JelleZijlstra I applied your suggested changes.
